ytmusicapi/ytmusic.py

```diff
@@ -81,7 +83,13 @@ def __init__(self,
         self.proxies = proxies
         self.cookies = {'CONSENT': 'YES+1'}
 
-        self.headers = prepare_headers(self._session, proxies, auth)
+        input_json = load_headers_file(self.auth)
+        self.input_dict = CaseInsensitiveDict(input_json)
+        self.is_oauth_auth = is_oauth(self.input_dict)
+        if self.is_oauth_auth:
+            self.ytmusic_oauth = YTMusicOAuth(self._session, self.proxies)
```

Comment:
This `if` should not be needed anymore if we change `prepare_headers` to take the `input_dict` as an argument.